### PR TITLE
Add delete event feature

### PR DESCRIPTION
**Pull request by Khaleelur-Rahman**
Implement delete event feature for events.

Note to self:
Add more test cases later

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/event/DeleteEventCommand.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_DELETE_EVENT_SUCCESS = "Deleted Event: %1$s";
+    private final Index targetIndex;
+    public DeleteEventCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
```
Comment (tjch-o) [inline src/main/java/seedu/address/logic/commands/event/DeleteEventCommand.java:32]: minor nits but I think there should be spacing between the `targetIndex` variable, constructor and the `@Override` annotation

File: src/main/java/seedu/address/logic/parser/event/DeleteEventCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.event.DeleteEventCommand;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.ParserUtil;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new DeleteEventCommand object
```
Comment (tjch-o) [inline src/main/java/seedu/address/logic/parser/event/DeleteEventCommandParser.java:12]: missing a full stop

**Review by tjch-o — CHANGES_REQUESTED** [review]
lgtm! minor nits to be tidied up before merging

File: src/main/java/seedu/address/model/event/UniqueEventList.java [hunk 3]
```diff
@@ -95,6 +113,28 @@ public void setEvents(List<Event> events) {
...
+                        updatedContactsList.add(p);
+                    }
+                }
+            } else {
+                updatedContactsList = contactsList;
+            }
+            Event updatedEvent = new Event(curr.getName(), curr.getDate(), curr.getAddress(), updatedContactsList);
+            internalList.set(i, updatedEvent);
+        }
+    }
```
Comment (tjch-o) [inline src/main/java/seedu/address/model/event/UniqueEventList.java:136]: ok now to think about it perhaps this one and the other `UpdateTag()` in `UniqueTagList.java` can be abstracted further to maintain SLAP but will KIV for now

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java
Comment (rionshocker) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:1]: minor nit but can consider having spacings to separate the cases to make it easier to see

**Review by rionshocker — APPROVED** [review]
LGTM! Just some minor changes to be considered.

File: src/main/java/seedu/address/logic/parser/event/DeleteEventCommandParser.java (hunk 2 above)
Comment (Khaleelur-Rahman) [reply src/main/java/seedu/address/logic/parser/event/DeleteEventCommandParser.java:12]: Good catch. Made the changes!

File: src/main/java/seedu/address/model/event/UniqueEventList.java (hunk 3 above)
Comment (Khaleelur-Rahman) [reply src/main/java/seedu/address/model/event/UniqueEventList.java:136]: yup, we'll look into it in 1.3

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java
Comment (Khaleelur-Rahman) [reply src/main/java/seedu/address/logic/parser/AddressBookParser.java:1]: Thanks! Made the necessary changes!

File: src/main/java/seedu/address/logic/commands/event/DeleteEventCommand.java (hunk 1 above)
Comment (Khaleelur-Rahman) [reply src/main/java/seedu/address/logic/commands/event/DeleteEventCommand.java]: good catch!

**Pull request merged**